save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/fossgraph/CreateVulnerabilityView.kt

```diff
@@ -17,23 +22,61 @@ import react.dom.html.ReactHTML.h1
 import react.dom.html.ReactHTML.label
 import react.dom.html.ReactHTML.span
 import react.dom.html.ReactHTML.textarea
+import react.router.useNavigate
 import react.useState
 
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+
 val createVulnerabilityView: VFC = VFC {
     useBackground(Style.BLUE)
 
+    val navigate = useNavigate()
+
     val (vulnerability, setVulnerability) = useState(VulnerabilityDto.empty)
-    val (vulnerabilityProjects, setVulnerabilityProjects) = useState(mutableListOf<VulnerabilityProjectDto>())
+    val (vulnerabilityProjects) = useState(mutableListOf<VulnerabilityProjectDto>())
+    val (deleteProject, setDeleteProject) = useState<VulnerabilityProjectDto?>(null)
 
     val projectWindowOpenness = useWindowOpenness()
-    val fetchProjectCredentials2: (VulnerabilityProjectDto) -> Unit = { project ->
+    val deleteProjectWindowOpenness = useWindowOpenness()
+
+    val fetchProject: (VulnerabilityProjectDto) -> Unit = { project ->
         vulnerabilityProjects.add(project)
+        setVulnerability(vulnerability.copy(projects = vulnerabilityProjects))
         projectWindowOpenness.closeWindow()
     }
 
+    val enrollRequest = useDeferredRequest {
+        val response = post(
+            url = "$apiUrl/vulnerabilities/save",
+            headers = jsonHeaders,
+            body = Json.encodeToString(vulnerability),
+            loadingHandler = ::loadingHandler,
+        )
+        if (response.ok) {
+            navigate(to = "/${FrontendRoutes.FOSS_GRAPH}")
+        }
+    }
+
+    displayModal(
+        deleteProjectWindowOpenness.isOpen(),
+        "Deletion of project",
+        "Are you sure you want to remove this project from the vulnerability?",
+        mediumTransparentModalStyle,
+        { deleteProjectWindowOpenness.closeWindow() },
```

Comment:
you can use here `com.saveourtool.save.frontend.utils.WindowOpenness#closeWindowAction`,
if it expects some argument, there is the extension: `com.saveourtool.save.frontend.utils.UtilsKt#withUnusedArg`

Reply:
Fixed